libs/import/src/lib/table-extracted-elements/contracts/contracts.component.ts

```diff
@@ -109,38 +111,40 @@ export class TableExtractedContractsComponent implements OnInit {
     }
   }
 
-  /**Verifies if terms overlap with existing mandate terms in the Db */
-  private async verifyTermOverlap(importState: ContractsImportState) {
-    const getTitleMandates = (ref: Query) => ref.where('type', '==', 'mandate')
-      .where('titleId', '==', importState.contract.titleId)
-      .where('status', '==', 'accepted');
-
-    const mandates = await this.contractService.valueChanges(getTitleMandates).pipe(
-      joinWith({
-        terms: (mandate: Mandate) => this.termService.valueChanges(mandate.termIds)
-      }, { shouldAwait: true }),
-      first()
-    ).toPromise() as FullMandate[];
+  async getExistingContracts(type: 'sale', titleId: string): Promise<FullSale[]>;
+  async getExistingContracts(type: 'mandate', titleId: string): Promise<FullMandate[]>;
+  async getExistingContracts(type: 'sale' | 'mandate', titleId: string): Promise<(FullSale | FullMandate)[]> {
+    const query = getTitleContracts(type, titleId);
+    const contracts = await this.contractService.getValue(query);
+    const promises = contracts.map(contract => this.termService.getValue(contract.termIds))
+    const terms = await Promise.all(promises);
+    return contracts.map((contract, idx) => ({ ...contract, terms: terms[idx] }) as FullSale | FullMandate)
+  }
 
-    return importState.terms.map(term => {
-      const availabilities = territoryAvailabilities(term, mandates, [])
-      //available territories indicates that there is an overlap with existing mandates in db.
-      return !!availabilities.available.length
-    }).some(isOverlapping => isOverlapping)
+  /**Verifies if terms overlap with existing mandate terms in the Db */
+  private async verifyOverlappingMandatesAndSales(importState: ContractsImportState) {
+    const mandates = await this.getExistingContracts('mandate', importState.contract.titleId);
+    const sales = await this.getExistingContracts('sale', importState.contract.titleId);
+    const availabilities = importState.terms.map(term =>
+      getOverlappingTerritoryAvailabilities(term, mandates, sales, [])
+      );
```

Comment:
If you do 3 lines, use the `return` syntax. Inlined syntax is only if you're doing it on one line
```suggestion
    const availabilities = importState.terms.map(term => {
      return getOverlappingTerritoryAvailabilities(term, mandates, sales, []);
    );
```